t_fault_injection/test_fault_injection_base.py

```diff
@@ -74,56 +78,97 @@ class TestFailFunctionBase(tester.TempestaTest):
-        # Remove function from fault injection list
-        cmd = "echo > /sys/kernel/debug/fail_function/inject"
+    def setup_fail_function_test(func_name, probability, times, space, retval):
+        # Write function name to special debug fs file.
+        cmd = f"echo {func_name} > /sys/kernel/debug/fail_function/inject"
+        out = remote.client.run_cmd(cmd)
+        # Write return code, which this function should return instead of
+        # real return code.
+        cmd = f"printf %#x {retval} > /sys/kernel/debug/fail_function/{func_name}/retval"
+        out = remote.client.run_cmd(cmd)
+        TestFailFunctionBase.__setup_common("fail_function", probability, times, space)
+
+    @staticmethod
+    def setup_fail_slab_test(task_filter, probability, times, space, ignore_gfp_wait):
+        # Setup task-filter.
+        cmd = f"echo {task_filter} > /sys/kernel/debug/failslab/task-filter"
+        out = remote.client.run_cmd(cmd)
+        # Inject fails into allocation that can sleep.
+        cmd = f"echo {ignore_gfp_wait} > /sys/kernel/debug/failslab/ignore-gfp-wait"
+        out = remote.client.run_cmd(cmd)
+        TestFailFunctionBase.__setup_common("fail_function", probability, times, space)
+
+    @staticmethod
+    def setup_fail_page_alloc_test(task_filter, probability, times, space, ignore_gfp_wait):
+        # Setup task-filter.
+        cmd = f"echo {task_filter} > /sys/kernel/debug/fail_page_alloc/task-filter"
+        out = remote.client.run_cmd(cmd)
+        # Inject fails into allocation that can sleep.
+        cmd = f"echo {ignore_gfp_wait} > /sys/kernel/debug/fail_page_alloc/ignore-gfp-wait"
         out = remote.client.run_cmd(cmd)
+        TestFailFunctionBase.__setup_common("fail_page_alloc", probability, times, space)
+
+    @staticmethod
+    def __teardown_common(failtype):
         # Restore probability
-        cmd = f"echo 0 > /sys/kernel/debug/fail_function/probability"
+        cmd = f"echo 0 > /sys/kernel/debug/{failtype}/probability"
         out = remote.client.run_cmd(cmd)
         # Restore interval
-        cmd = f"echo 1 > /sys/kernel/debug/fail_function/interval"
+        cmd = f"echo 1 > /sys/kernel/debug/{failtype}/interval"
         out = remote.client.run_cmd(cmd)
         # Restore times
-        cmd = f"echo 1 > /sys/kernel/debug/fail_function/times"
+        cmd = f"printf %#x 1 > /sys/kernel/debug/{failtype}/times"
         out = remote.client.run_cmd(cmd)
         # Restore space
-        cmd = f"echo 0 > /sys/kernel/debug/fail_function/space"
+        cmd = f"echo 0 > /sys/kernel/debug/{failtype}/space"
         cmd = remote.client.run_cmd(cmd)
 
+    @staticmethod
+    def teardown_fail_function_test():
+        # Remove function from fault injection list
+        cmd = f"echo > /sys/kernel/debug/fail_function/inject"
+        out = remote.client.run_cmd(cmd)
+        TestFailFunctionBase.__teardown_common("fail_function")
+
+    @staticmethod
+    def teardown_fail_slab():
```

Comment:
`teardown_fail_slab()/setup_fail_slab_test()` and `teardown_fail_page_alloc()/setup_fail_page_alloc_test()` are not used please remove them.